test/jasmine/tests/calcdata_test.js

```diff
@@ -869,6 +871,216 @@ describe('calculated data and points', function() {
+
+            function makeData(type, a, b, axId) {
+                var input = [a, b];
+                var cat = input[axId === 'yaxis' ? 1 : 0];
+                var data = input[axId === 'yaxis' ? 0 : 1];
+
+                var measure = [];
+                for(j = 0; j < data.length; j++) {
+                    measure.push('absolute');
+                }
+
+                var z = Lib.init2dArray(cat.length, data.length);
+                for(j = 0; j < z.length; j++) {
+                    for(k = 0; k < z[j].length; k++) {
+                        z[j][k] = 0;
+                    }
+                }
+                if(axId === 'xaxis') {
+                    for(j = 0; j < b.length; j++) {
+                        z[0][j] = b[j];
+                    }
+                }
+                if(axId === 'yaxis') {
+                    for(j = 0; j < b.length; j++) {
+                        z[j][0] = b[j];
+                    }
+                }
+
+                return Lib.extendDeep({}, {
+                    orientation: axId === 'yaxis' ? 'h' : 'v',
+                    type: type,
+                    x: cat,
+                    a: cat,
+
+                    b: data,
+                    y: data,
+                    z: z,
+
+                    // For OHLC
+                    open: data,
+                    close: data,
+                    high: data,
+                    low: data,
+
+                    // For waterfall
+                    measure: measure,
+
+                    // For splom
+                    dimensions: [
+                        {
+                            label: 'DimensionA',
+                            values: a
+                        },
+                        {
+                            label: 'DimensionB',
+                            values: b
+                        }
+                    ]
+                });
+            }
+
+            supportedCartesianTraces.forEach(function(trace) {
+                ['xaxis', 'yaxis'].forEach(function(axId) {
+                    if(axId === 'yaxis' && oneOrientationTraces.indexOf(trace.type) !== -1) return;
+                    ['value ascending', 'value descending'].forEach(function(categoryorder) {
+                        it('sorts ' + axId + ' by ' + categoryorder + 'for trace type ' + trace.type, function(done) {
+                            var data = [7, 2, 3];
+                            var baseMock = { data: [makeData(trace.type, cat, data, axId)], layout: {}};
+                            var mock = Lib.extendDeep({}, baseMock);
+                            mock.layout[axId] = { type: 'category', categoryorder: categoryorder};
+
+                            // Set ordering
+                            var finalOrder = ['b', 'c', 'a'];
+                            if(categoryorder === 'value descending') finalOrder.reverse();
+
+                            if(trace.type.match(/histogram/)) {
```

Comment:
Could we reuse `checkAggregatedValue` here?

Reply:
Absolutely! I also realized that very first test was very similar to a subsequent one so I :hocho: it.

See for 2d379f7647bec909b806f56276b6f6225578bc6b for the changes!